src/main/java/com/PBA/authservice/facade/UserFacadeImpl.java

```diff
@@ -46,4 +50,21 @@ public UserDto getUser(UUID uid) {
         UserProfileDto userProfileDto = activeUserMapper.toUserProfileDto(activeUserProfile);
         return activeUserMapper.toUserDto(activeUser, userProfileDto);
     }
+
+    @Override
+    public UserDto verifyUser(UUID validationCode) {
+        PendingUser userToValidate = pendingUserService.getPendingUserByValidationCode(validationCode);
+        userValidator.validatePendingUser(userToValidate);
+        ActiveUser activeUser = pendingUserMapper.toActiveUser(userToValidate);
+
+        PendingUserProfile pendingUserProfile = pendingUserService.getPendingUserProfileByUserId(userToValidate.getId());
```

Comment:
When you see that you tend to add a new line to separate the code, for sure that code should be extracted in a separate private method.

Try to make this function a little smaller using new functions.